test-harness/init.sh

```diff
@@ -77,17 +78,21 @@ function template_build_targets() {
     [[ -z $GIT_DIFF_SOURCEBRANCH ]] && echo "ERROR: GIT_DIFF_SOURCEBRANCH wasn't provided" && return 1
 
     echo "INFO: Running git diff from branch ${GIT_DIFF_SOURCEBRANCH}"
-    files=(`git diff ${GIT_DIFF_UPSTREAMBRANCH} ${GIT_DIFF_SOURCEBRANCH} --name-only|grep -v *.md`)
+    files=(`git diff ${GIT_DIFF_UPSTREAMBRANCH} ${GIT_DIFF_SOURCEBRANCH} --name-only|grep -v *.md||true`)
     for file in "${files[@]}"
     do
         IFS='/' read -a folder_array <<< "${file}"
-        
         if [ ${#folder_array[@]} -lt 3 ]; then
             continue
         fi
 
+        if [ ${folder_array[0]}=='test-harness' ]; then 
```

Comment:
Can you explain the special index values of `0` and `1`? How do we know that these are the right indices to look for these folder names?

Reply:
sure... So if you think about the folder parts of the `PWD`. If any file(s) happen to change within `/test-harness` or `/infra/modules` then the harness runs all tests within the `/infra/templates` should be carried over to the build directory for the run via `default_to_all_template_paths`. It's a naive approach and we have a user story to design a more elegant algo based off a module dependency graph.

Reply:
Makes sense, thanks for clarifying